src/cli.ts

```diff
@@ -33,67 +44,229 @@ export default function run(): void {
     process.exit(1);
   }
 
-  if (!commander.args[0]) {
+  if (!commander.args[0] && !commander.project) {
     console.error("Source directory is required.");
     process.exit(1);
   }
 
-  const outDir = commander.outDir;
-  const srcDir = commander.args[0];
-
   const options: CLIOptions = {
+    outDirPath: commander.outDir,
+    srcDirPath: commander.args[0],
+    project: commander.project,
     outExtension: commander.outExtension,
     excludeDirs: commander.excludeDirs ? commander.excludeDirs.split(",") : [],
     quiet: commander.quiet,
     sucraseOptions: {
-      transforms: commander.transforms.split(","),
+      transforms: commander.transforms ? commander.transforms.split(",") : [],
       enableLegacyTypeScriptModuleInterop: commander.enableLegacyTypescriptModuleInterop,
       enableLegacyBabel5ModuleInterop: commander.enableLegacyBabel5ModuleInterop,
       jsxPragma: commander.jsxPragma || "React.createElement",
       jsxFragmentPragma: commander.jsxFragmentPragma || "React.Fragment",
     },
   };
 
-  buildDirectory(srcDir, outDir, options).catch((e) => {
+  buildDirectory(options).catch((e) => {
     process.exitCode = 1;
     console.error(e);
   });
 }
 
-async function buildDirectory(
-  srcDirPath: string,
-  outDirPath: string,
-  options: CLIOptions,
-): Promise<void> {
+interface FileInfo {
+  srcPath: string;
+  outPath: string;
+}
+
+async function findFiles(options: CLIOptions): Promise<Array<FileInfo>> {
+  const outDirPath = options.outDirPath;
+  const srcDirPath = options.srcDirPath;
+
   const extensions = options.sucraseOptions.transforms.includes("typescript")
     ? [".ts", ".tsx"]
     : [".js", ".jsx"];
+
   if (!(await exists(outDirPath))) {
     await mkdir(outDirPath);
   }
+
+  const outArr: Array<FileInfo> = [];
   for (const child of await readdir(srcDirPath)) {
     if (["node_modules", ".git"].includes(child) || options.excludeDirs.includes(child)) {
       continue;
     }
     const srcChildPath = join(srcDirPath, child);
     const outChildPath = join(outDirPath, child);
     if ((await stat(srcChildPath)).isDirectory()) {
-      await buildDirectory(srcChildPath, outChildPath, options);
+      const innerOptions = {...options};
+      innerOptions.srcDirPath = srcChildPath;
+      innerOptions.outDirPath = outChildPath;
+      const innerFiles = await findFiles(innerOptions);
+      outArr.push(...innerFiles);
     } else if (extensions.some((ext) => srcChildPath.endsWith(ext))) {
       const outPath = outChildPath.replace(/\.\w+$/, `.${options.outExtension}`);
-      await buildFile(srcChildPath, outPath, options);
+      outArr.push({
+        srcPath: srcChildPath,
+        outPath,
+      });
+    }
+  }
+
+  return outArr;
+}
+
+async function runGlob(options: CLIOptions): Promise<Array<FileInfo>> {
+  const tsConfigPath = join(options.project, "tsconfig.json");
+
+  let str;
+  try {
+    str = await readFile(tsConfigPath, "utf8");
+  } catch (err) {
```

Comment:
`exists` is deprecated ( https://nodejs.org/api/fs.html#fs_fs_exists_path_callback ). The documentation recommends just trying the `readFile` and handling the error instead of the `exists` + `readFile` since that's a timing race condition.

Reply:
Interesting, fair enough. If possible, it would be nice to detect if it's a "file not found" error (maybe `err.code === 'ENOENT'` like in the docs) and re-throw if it's an unrecognized error.

(In this case, I'm not convinced that the race condition actually matters, since the worst case is an uglier error message, and there are plenty of similar cross-file race conditions that can't be reasonably handled, but I guess it's nice to make the ugly error impossible.)